src/graph/tuning.cc

```diff
@@ -117,24 +117,26 @@ ncclResult_t ncclTopoTuneModel(struct ncclComm* comm, int minCompCap, int maxCom
     // Skipping on SCCL algorithms here since SCCL tune the algorithm in the synthesizer.
 
     for (int a=0; a<NCCL_NUM_ALGORITHMS; a++) {
-      if (coll == ncclFuncAllToAll || a == NCCL_ALGO_SCCL) {
-        // SCCL algorithm has hooks for AllToAll, AllGather, AllReduce and ReduceScatter
+      if (coll == ncclFuncAllToAll || coll == ncclFuncCustomCollective || a == NCCL_ALGO_SCCL) {
+        // SCCL algorithm has hooks for AllToAll, AllGather and ReduceScatter
         // SCCL algorithm is dynamic and busBw/latency can only be determined by the input XML algorithm. An analysis will be added later.
         for (int p=0; p<NCCL_NUM_PROTOCOLS; p++) {
-          if ((coll == ncclFuncAllToAll || coll == ncclFuncAllGather || coll == ncclFuncReduceScatter || coll == ncclFuncAllReduce) && a == NCCL_ALGO_SCCL && p == comm->scclAlgo.protocol){
+          if ((coll == ncclFuncAllToAll || coll == ncclFuncAllGather || coll == ncclFuncReduceScatter) && a == NCCL_ALGO_SCCL && p == comm->scclAlgo.protocol) {
             // Setting the bandwidth and latency values to 1.0 (some arbitrary value) so that they don't get skipped by ncclTopoGetAlgoTime
             comm->bandwidths[coll][a][p] = 1.0;
             comm->latencies[coll][a][p] = 1.0;
           } else {
+            //Set all protocols for NCCL_ALGO_SCCL for ncclFuncCustomCollective to 0 because XML is loaded later.
             comm->bandwidths[coll][a][p] = 0.0; // This will make sure that sccl is not selected for any other scenario
             comm->latencies[coll][a][p] = 0.0;
           }
         }
         continue;
       }
 
+      
       if (coll != ncclFuncAllReduce && a != NCCL_ALGO_RING) continue;
-
+      if (coll == ncclFuncCustomCollective) continue;
```

Comment:
this should not be necessary based on tuning parameters set earlier, right? There is only one customCollective algorithm and that only one should have a > 0  bw